src/v/datalake/schema_avro.cc

```diff
@@ -10,8 +10,10 @@
 #include "datalake/schema_avro.h"
 
 #include <seastar/util/defer.hh>
+#include <seastar/util/log.hh>
```

Comment:
So some code which used logging never included the header but got it indirectly from `vassert.h`. This has been removed from vassert.h (since no logging is done in header) so we add the header where it is needed.